### DISCUSSION TITLE
Performance Issues with Large Rnote Documents

### DISCUSSION THREAD

**Discussion by xCheWy88** [flxzt/rnote#1122] [Development]
Hi Rnote Development Team,

First, I want to express my gratitude for creating such an excellent note-taking software. I use Rnote religiously for my studies, and it has become an essential tool in my academic life.

However, I have encountered a significant performance issue that hampers my workflow. When I accumulate a substantial amount of content within a single Rnote document, the software becomes extremely laggy and unresponsive. This happens despite my PC having a decent setup, which leads me to believe that the issue might be related to the software itself rather than my hardware.

Here are the details of my setup:
Processor: i5 10400
RAM: 16gb
Storage: ssd
Operating System: win11 latest
    
I would like to know if there are any plans to improve the software's performance with large documents or if there are any recommended best practices to mitigate this issue. Any advice or insights would be greatly appreciated, as Rnote is an indispensable part of my study routine.

Thank you for your hard work and dedication to improving Rnote!

Best regards

**Comment by flxzt** [comment]
Hi, thanks for the kind words.

What kind of performance problems are you experiencing, can you go into more detail? For example, does it lag when a lot of strokes are in the current view or in general when there are a lot of strokes on the loaded document.
Can you determine if the RAM or the CPU usage is the limiting factor?

Thanks!

**Reply by xCheWy88** [reply]
So I use it a lot for creating maps or concept maps to learn. Mostly using the type or insert text function and occasionally shapes and some lines drawn with my xp pen.

Details of lag:

https://github.com/flxzt/rnote/assets/131624205/fff4146a-ddce-4d1b-8441-1236be8e82b6


After a certain point when there's a lot of content or text in the rnote document, when I insert a text or even move the page around or move the selected text or selected object such as an image etc. (basically most editing) will start to become very laggy. I have done a screen recording for the example of just moving or dragging the page, when I click, hold and move(drag) the page, the shaking of the pointer or cursor is to show how long the page actually moves or responds, the same applies to when selecting objects or texts.

![Picture1](https://github.com/flxzt/rnote/assets/131624205/278e9a5d-9ff1-475b-8c3a-d4765ea2a260)

It could be the CPU potentially, perhaps I can send you a sample file to you via email to see if it occurs with your setup, assuming that it has better specs?
